### ISSUE TITLE
Odd-looking slices for fields created using add_sph_fields

### ISSUE THREAD

**Issue by jzuhone** [yt-project/yt#2415]
**Bug summary**

The `add_sph_fields` functionality allows one to take non-gas particles and do visualizations with them in yt-4.0. Currently, the results are somewhat odd for `SlicePlot` objects (see linked notebook below for an example). I've tracked this down to the fact that by default there is a normalization step applied when creating a slice:

https://github.com/yt-project/yt/blob/yt-4.0/yt/geometry/coordinates/cartesian_coordinates.py#L343-L373

This appears to work fine for normal SPH fields, but not for fields created using `add_sph_fields`. If one sets `ds.use_sph_normalization = False`, then the results look more sensible, but I am not sure if this is what we really want to do. 

I'm actually a bit lost on what the normalization is doing, so if someone could clear this up that'd be great. I think maybe @ngoldbaum or @qobilidop may know something about this?

**Code for reproduction**

A full example of what things look like when `ds.use_sph_normalization = True` and when `ds.use_sph_normalization = False` is shown here:

https://gist.github.com/11db3573f3bd589fd04ce970a02db94f

**Comment by ngoldbaum** [comment]
I think the justification for using the normalization is more or less that it looks nice in a lot of cases and enhances detail. However in cases when the individual smoothing regions might be visible, it does poorly. Quoting the SPLASH paper:

> The normalised interpolation performs poorly at the edges, where the effects of individual particle smoothing spheres
are visible. However, using a normalised interpolation improves the accuracy of volume rendered quantities on the pixels by removing effects due to the particle distribution. Thus it is recommended that a normalised interpolation should always be used if there are no free surfaces.

How many nearest neighbor particles does the interpolation method in `add_sph_field` use to generate smoothing lengths? Maybe that number needs to increase?

Do the slices look better with gather smoothing?

**Comment by jzuhone** [comment]
@ngoldbaum you're right, the slices do look much more sensible with gather smoothing. I'm going to mark this as `wontfix` and remove the `bug` tag.

**Comment by jzuhone** [comment]
Actually, I was wrong--gather smoothing yields the same result.

**Comment by jzuhone** [comment]
And increasing `n_neighbors=64` yields no improvement.

**Comment by matthewturk** [comment]
Does it change the appearance at all?
